packages/cactus-plugin-ledger-connector-corda/src/main/typescript/plugin-ledger-connector-corda.ts

```diff
@@ -243,4 +322,169 @@
   public async getFlowList(): Promise<string[]> {
     return ["getFlowList()_NOT_IMPLEMENTED"];
   }
+
+  private async setupRequest(
+    username: string,
+    password: string,
+  ): Promise<{ headers: any; agent: any }> {
+    const authString = Buffer.from(`${username}:${password}`).toString(
+      "base64",
+    );
+    const headers = { Authorization: `Basic ${authString}` };
+    const rejectUnauthorized = process.env.REJECT_UNAUTHORIZED !== "false";
+    const httpsAgent = new https.Agent({
+      rejectUnauthorized,
+    });
+    return { headers, agent: httpsAgent };
+  }
+
+  public async startFlow(req: StartFlowV1Request): Promise<any> {
+    try {
+      const { headers, agent } = await this.setupRequest(
+        req.username,
+        req.password,
+      );
+      const holdingIDShortHash = req.holdingIDShortHash;
+      const cordaReq = {
+        clientRequestId: req.clientRequestId,
+        flowClassName: req.flowClassName,
+        requestBody: req.requestBody,
+      };
+      const cordaReqBuff = Buffer.from(JSON.stringify(cordaReq));
+      const response = await fetch(`${req.baseUrl}flow/${holdingIDShortHash}`, {
+        method: `POST`,
+        headers,
+        body: cordaReqBuff,
+        agent,
+      });
```

Comment:
@petermetz the URL's are the input corresponding to the REST api calls (which will be a variable), so I think its a false positive CodeQL review point

Reply:
@jagpreetsinghsasan If I send in a request with `baseUrl` being set to `http://evil.website.with.malware.example.com` then we'll go to that link and bad things could happen.

Is there any reason why we can't pass in the hostname of the Corda node(s) in the constructor of the connector plugin? That would make it less flexible but then we only have to worry about a malicious system administrator who deployed the system with a bad configuration (an acceptable risk because if the person deploying the software is malicious, all is lost anyway)

Reply:
hi @petermetz, am currently working on the changes to address the codeql issues. I think the changes I'm doing will be able to handle this concern.